opal/corelib/enumerable.rb

```diff
@@ -177,6 +177,10 @@ def count(object = undefined, &block)
   end
 
   def cycle(n = nil, &block)
+    if `arguments.length > 1`
```

Comment:
Seems to impact a lot of cases where there is a default parameter value (not just blocks):

```ruby
def stuff(foo=123)
end
```

```js
/* Generated by Opal 0.9.0.beta1 */
(function(Opal) {
  Opal.dynamic_require_severity = "error";
  var OPAL_CONFIG = { method_missing: true, arity_check: true, freezing: true, tainting: true };
  var self = Opal.top, $scope = Opal, nil = Opal.nil, $breaker = Opal.breaker, $slice = Opal.slice;

  return (Opal.defn(Opal.Object, '$stuff', function(foo) {var $arity = arguments.length;if ($arity < 0) { Opal.ac($arity, -1, this, "stuff"); }
    var self = this;

    if (foo == null) {
      foo = 123
    }
    return nil;
  }), nil) && 'stuff'
})(Opal);

/* Generated by Opal 0.9.0.beta1 */
(function(Opal) {
  Opal.dynamic_require_severity = "error";
  var OPAL_CONFIG = { method_missing: true, arity_check: true, freezing: true, tainting: true };
  var self = Opal.top, $scope = Opal, nil = Opal.nil, $breaker = Opal.breaker, $slice = Opal.slice;

  Opal.add_stubs(['$exit']);
  return $scope.get('Kernel').$exit()
})(Opal);

```